.github/workflows/macos.yml

```diff
@@ -49,28 +51,44 @@ jobs:
         run: |
           mkdir _build
           echo "NPROC=`sysctl -n hw.ncpu`" >> $GITHUB_ENV
+          export TERM="vt100"
+          echo "TERM=$TERM" >> $GITHUB_ENV
 
       - name: configure
         run: |
           cd _build
-          ../configure --enable-cobc-internal-checks --enable-hardening
+          ../configure --enable-cobc-internal-checks --enable-debug --prefix /opt/cobol/gnucobol-gcos --exec-prefix /opt/cobol/gnucobol-gcos
 
       - name: Upload config.log
         uses: actions/upload-artifact@v3
-        if: failure()
         with:
           name: config.log
           path: _build/config.log
+#       if: failure()
 
       - name: make
         run: |
           cd _build
           make --jobs=$((${NPROC}+1))
 
+# make install must be done before make check, otherwise execution of
+# generated COBOL files fail for a missing /usr/local/lib/libcob.dylib
```

Comment:
That's cheating :-) But ok. Still: The question is how to make it work without that.

Reply:
To be honest, I have not yet completely understood how tests are ran, I have only a vague idea of how `autoconf` and other tools are working. So yes, I decided to cheat a little :-)

Reply:
minimal overview:
* autoconf compiles the testsuite source to a script that is executable anywhere and all self-contained in the generated script
* this script first sources atconfig and atlocal (both generated when running the configure script) which sets up the necessary "local environment" to get this running from the build tree
* to make this work our atlocal.in is quite huge, resets all GnuCOBOL related environment variables (you don't want to trash anything in the production and) and sets only some, like where to find the configuration and which options have to be passed by cobc to the linker and used by the environment to execute the resulting binaries (and this is the part where macos does not work because of "security features", which _should_ be possible the handle via linker options)